numpy/compat/py3k.py

```diff
@@ -104,7 +100,11 @@ def is_pathlib_path(obj):
 
     Prefer using `isinstance(obj, os_PathLike)` instead of this function.
     """
-    return Path is not None and isinstance(obj, Path)
+    from pathlib import Path
+    from warnings import warn
+    warn("The use of is_pathlib_path is deprecated. "
+         "Prefer using `isinstance(obj, os_PathLike)`.", stacklevel=2)
```

Comment:
Deprecations need a release note, but otherwise this is totally fair game to deprecate.